src/stripe/objects/core/charge.cr

```diff
@@ -78,44 +78,68 @@ class Stripe::Charge
 
   getter id : String
   getter amount : Int32
+  getter amount_captured : Int32?
   getter amount_refunded : Int32
   getter application : String?
   getter application_fee : String?
-  getter balance_transaction : String?
+  getter application_fee_amount : Int32?
+  getter balance_transaction : String? # Expandable to Stripe::BalanceTransaction
+
+  class BilingDetails
+    include JSON::Serializable
+
+    getter address : Stripe::Shipping::Address
+    getter email : String?
+    getter name : String?
+    getter phone : String?
+  end
+
+  getter billing_details : BilingDetails?
+
   getter captured : Bool?
 
   @[JSON::Field(converter: Time::EpochConverter)]
   getter created : Time
 
   getter currency : String
-  getter customer : String?
+  getter customer : String? | Stripe::Customer?
   getter description : String?
   getter destination : String?
   getter dispute : String?
   getter failure_code : String?
   getter failure_message : String?
   getter fraud_details : FraudDetails?
-  getter invoice : String?
+  getter invoice : String? | Stripe::Invoice?
   getter livemode : Bool
   getter metadata : Hash(String, String)?
-  getter on_behalf_of : String?
-  getter order : String?
+  getter on_behalf_of : String? # Expandable to Stripe::Account
```

Comment:
I'm definitely not against a comment, but why not something like `| Stripe::Account?`? I'm sure there's a reason I'm not seeing, just curious!

Reply:
You mean like # | Stripe::Account?
I can do that (I thought it was more explicit this way)

Reply:
Oh, I thought `Stripe::Account` was a class this shard already defined!

So what do you mean when you say "Expandable to Stripe::Account"? That a user of this shard could take the String here and map it to a custom class they write?

Reply:
Oh, I thought `Stripe::Account` was a class this shard already defined!

So what do you mean when you say "Expandable to Stripe::Account"? That a user of this shard could take the String here and map it to a custom class they write?

Reply:
I mean when Stripe::Account will exist (I'm working on it), a stripe.cr user will be able to call `Stripe::Charge.retrieve({
  id: "cs_test_KdjLtDPfAjT1gq374DMZ3rHmZ9OoSlGRhyz8yTypH76KpN4JXkQpD2G0",
  expand: ["on_behalf_of"],
})`  and get a `Stripe::Account` object in `charge.on_behalf_of` instead of a string (in this case, the id).
More here : https://stripe.com/docs/expand

Reply:
Ah, I understand!

Yeah, in that case my preference would be zero comments, and just make some notes on the issue/pull request for `Stripe::Account` to come back in and add explicit `| Stripe::Account?` code.

Either that, or we could more explicitly label it with `TODO` since that seems to be more of what this comment's job is for now. I'm not sure if you have a code style preference on this kind of stuff, @confact?

Reply:
I have added support for expand in the subscription. It handles more or less auto by doing `| Stripe.:Account` on `on_behalf_of` - And add support to set expand, of course.

You can have like this for now and fix expand when we have Stripe::Account.

Reply:
I have some ideas about the expand capability : we could annotate the expandable fields with `@[ Expandable]` so that the API methods (retrieve, create, list ...) can check if the `expand` string arguments are valid. I won't implement this here, but would be subject for a PR :-D

Anyway, is there anything to change/add to this PR ?

Reply:
@rmarronnier maybe add support for the Account now when it got merged? Otherwise, it looks good!